data/simpy.gram

```diff
@@ -114,90 +110,90 @@ dotted_name[expr_ty]:
         _Py_Try(b, ex, el, f, EXTRA) }
-    | a='try' ':' b=block ex=except_block+ el=else_block {
+    | 'try' ':' b=block ex=except_block+ el=else_block {
         _Py_Try(b, ex, el, NULL, EXTRA) }
-    | a='try' ':' b=block ex=except_block+ f=finally_block {
+    | 'try' ':' b=block ex=except_block+ f=finally_block {
         _Py_Try(b, ex, NULL, f, EXTRA) }
-    | a='try' ':' b=block ex=except_block+ {
+    | 'try' ':' b=block ex=except_block+ {
         _Py_Try(b, ex, NULL, NULL, EXTRA) }
-    | a='try' ':' b=block f=finally_block {
+    | 'try' ':' b=block f=finally_block {
         _Py_Try(b, NULL, NULL, f, EXTRA) }
 except_block[excepthandler_ty]:
-    | a='except' e=expression 'as' t=target ':' b=block {
+    | 'except' e=expression 'as' t=target ':' b=block {
         _Py_ExceptHandler(e, t->v.Name.id, b, EXTRA) }
-    | a='except' e=expression ':' b=block {
+    | 'except' e=expression ':' b=block {
         _Py_ExceptHandler(e, NULL, b, EXTRA) }
-    | a='except' ':' b=block {
+    | 'except' ':' b=block {
         _Py_ExceptHandler(NULL, NULL, b, EXTRA) }
 finally_block[asdl_seq*]: 'finally' ':' a=block { a }
 
 return_stmt[stmt_ty]:
-    | a='return' b=expressions { _Py_Return(b, EXTRA) }
-    | a='return' { _Py_Return(NULL, EXTRA) }
+    | 'return' a=expressions { _Py_Return(a, EXTRA) }
+    | 'return' { _Py_Return(NULL, EXTRA) }
 
 raise_stmt[stmt_ty]:
-    | a='raise' b=expression 'from' c=expression { _Py_Raise(b, c, EXTRA) }
-    | a='raise' b=expression { _Py_Raise(b, NULL, EXTRA) }
-    | a='raise' { _Py_Raise(NULL, NULL, EXTRA) }
+    | 'raise' a=expression 'from' b=expression { _Py_Raise(a, b, EXTRA) }
+    | 'raise' a=expression { _Py_Raise(a, NULL, EXTRA) }
+    | 'raise' { _Py_Raise(NULL, NULL, EXTRA) }
 
 function_def[stmt_ty]:
     | d=decorators f=function_def_raw { function_def_decorators(p, d, f) }
     | function_def_raw
 
 function_def_raw[stmt_ty]:
-    | s=ASYNC 'def' n=NAME '(' params=parameters ')' a=['->' z=annotation { z }] ':' b=block {
+    | ASYNC 'def' n=NAME '(' params=parameters ')' a=['->' z=annotation { z }] ':' b=block {
         _Py_AsyncFunctionDef(((expr_ty) n)->v.Name.id, params, b, NULL, a, NULL, EXTRA) }
-    | s=ASYNC 'def' n=NAME '(' ')' a=['->' z=annotation { z }] ':' b=block {
+    | ASYNC 'def' n=NAME '(' ')' a=['->' z=annotation { z }] ':' b=block {
         _Py_AsyncFunctionDef(((expr_ty) n)->v.Name.id, empty_arguments(p), b, NULL, a, NULL, EXTRA) }
-    | s='def' n=NAME '(' params=parameters ')' a=['->' z=annotation { z }] ':' b=block {
+    | 'def' n=NAME '(' params=parameters ')' a=['->' z=annotation { z }] ':' b=block {
         _Py_FunctionDef(((expr_ty) n)->v.Name.id, params, b, NULL, a, NULL, EXTRA) }
-    | s='def' n=NAME '(' ')' a=['->' z=annotation { z }] ':' b=block {
+    | 'def' n=NAME '(' ')' a=['->' z=annotation { z }] ':' b=block {
         _Py_FunctionDef(((expr_ty) n)->v.Name.id, empty_arguments(p), b, NULL, a, NULL, EXTRA) }
```

Comment:
I think these could be collapsed with judicious use of `params ? params : empty_arguments(p)`. Ditto for the ASYNC version above.